packages/headless/src/auth/index.html

```diff
@@ -0,0 +1,37 @@
+<html>
+  <head>
+    <script type="module">
+      import {buildSamlClient} from './saml-client';
+
+      const organizationId = 'pvnh4ja7n5y7lmgstt3fpwy2bzq';
+
+      const saml = buildSamlClient({
+        organizationId,
+        provider: 'oktaA323aab78b9f1-45b5-a095-a1f0fa09ddd5',
+      });
+
+      async function main() {
+        const accessToken = await saml.authenticate();
+
+        buildSearchEngine({
+          configuration: {
+            organizationId,
+            accessToken,
+            renewAccessToken: () => saml.authenticate(),
+          },
+        });
+
+        // another engine
+        buildSearchEngine({
+          configuration: {
+            organizationId,
+            accessToken,
+            renewAccessToken: () => saml.authenticate(),
+          },
+        });
+      }
+
```

Comment:
Here's what configuring SAML would look like. I opted to not use middleware in the first version, because it would add significant complexity since there can be multiple engine instances. One would need to

- Share the `accessToken` somehow (e.g. cookies, local storage, window event listener?) to avoid each instance from needing to redirect.
- Coordinate multiple instances to not race to handshake. One would handshake, while others wait and read the file result.


With the current example, using only a single SAML client instance simplifies a lot of the complexity, while still being easy to configure.